eclipse-theia/theia
# Disable 'Save As...' in Electron when no file is selected

#### What it does
This fixes a problem in our Electron application where the 'Save As...' menu item had disappeared.  With this change, the visibility and enablement of all URI-based Electron menu items now update correctly as the selection changes.

These changes do not cover all commands and handlers where visibility and enablement depends on dynamic state such as the workspace selection.  Only commands that use UriAwareCommandHandler will have their menu state updated.

#### How to test
Start the Electron example.  Select a file and check that 'Save As...' menu is visible.  Now de-select the file, so there is no selection, and you should see 'Save As...' menu is not visible.  You can test for disabled menu items and other permutations by editing code in 'isVisible' and 'isEnabled' in URL based handlers.

- Check the browser menus still work as before
- Check Electron context menus work as before
- Check 'when' is still being picked up from the context

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
